common/src/main/java/org/conscrypt/ProvidedSessionDecorator.java

```diff
@@ -18,17 +18,23 @@
 
 import java.security.Principal;
 import java.security.cert.Certificate;
+import java.util.HashMap;
 import java.util.List;
 import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSessionBindingEvent;
+import javax.net.ssl.SSLSessionBindingListener;
 import javax.net.ssl.SSLSessionContext;
 import javax.security.cert.X509Certificate;
 
 /**
  * A {@link SessionDecorator} that externalizes the provider of the delegate session. This allows
- * the underlying session to be changed externally.
+ * the underlying session to be changed externally.  This class implements the application-level
```

Comment:
@flooey Maybe we should rename this class to something like `ExternalSession` or `PublicSession` to make the purpose more clear.  WDYT?

Reply:
@nmittler Yeah, I like that.  Will do.